classes/class-pmpro-subscription.php

```diff
@@ -981,30 +981,16 @@ public function save() {
 			return false;
 		}
 
-		// TODO: Perform validation here. The setter should make sure that fields have valid values,
-		// but we should also check here that the values make sense together. For example, cancelled
-		// subscriptions should not have a next payment date and startdates should not be before end dates.
-
-		$create = empty( $this->id );
-
-		if ( $create ) {
```

Comment:
@dparker1005 Did we intend to remove hooks here?

Reply:
I took it out in case we wanted to rethink this logic. How this was originally coded is how our save order function works, but I'm not sure how frequently those hooks are ever used. What seems more useful to me is how WC uses hooks around subscription status changes where you can hook into a subscription changing to active, changing to canceled, etc. So I removed this just so that we're not committed to a single hook approach and we can discuss exactly what hooks we want/how they should work instead of just blindly adding them